evalml/automl/automl_search.py

```diff
@@ -532,7 +532,7 @@ def search(self, data_checks="auto", show_iteration_plot=True):
             X_threshold_tuning = None
             y_threshold_tuning = None
             X_train, y_train = self.X_train, self.y_train
-            if self.optimize_thresholds and self.objective.is_defined_for_problem_type(ProblemTypes.BINARY) and self.objective.can_optimize_threshold:
+            if self.optimize_thresholds and self.objective.is_defined_for_problem_type(ProblemTypes.BINARY) and self.objective.can_optimize_threshold and self.problem_type == ProblemTypes.BINARY:
```

Comment:
@bchen1116 can you please use the new `is_binary` helper in `evalml/problem_types/utils.py`? That way, this will be true for timeseries binary classification too